docs/fhir-hl7v2-converter/configuration.md

```diff
@@ -0,0 +1,167 @@
+strings until they are used in the FHIR Path expression of an element
+2. resource - the focus resource is set and can be used in the subsequent properties. If the resource evaluates to empty
+then the element is not processed.
+3. condition - if the condition evaluates to false then the element is not processed
+4. required - if required is true and condition is false then an exception is thrown
+5. value and hl7Spec - if the element is to set a value
+6. schema and resourceIndex - if the element is a child schema
+
+# About Constants
+Constants provide great flexibility to generate reusable schemas and handle repeating types and fields, but with great 
+flexibility can also come great headaches and care must be taken on how constants are used.
+
+1. Care must be taken not to use FHIR Path syntax in an HL7 v2 location spec. It is recommended to use specific constants
+for FHIR Path and others for Hl7 v2 location specs to reduce the risk of a syntax error.
+2. Simple integer constants, like a resource index, can be used in both FHIR Path expressions and HL7 Specs
+(e.g. `myIndex: 1`)
+3. Constants are treated as strings until they are used on a specific element. At that time, if a constant is used
+in a FHIR Path expression then the contents of the string are evaluated as a FHIR Path expression for the result to 
+be inserted as replacement for the constant. This allows you to define constants with FHIR Path expressions
+that can be reused, but that the result will be dependent on what constants and focus resource is set on the element using it.
+4. Constants used as literals in FHIR Path expressions need to be surrounded by quotes (e.g. `myConstant: '"my literal string"'`)
+
+Constants are used as follows:
+* **For FHIR Path expressions** please refer to the
+[FHIR Path specification section on environment variables](http://hl7.org/fhirpath/N1/#environment-variables).
+Constants must have proper FHIR Path expression syntax including for string literals. This library also supports
+having constants as a prefix by using the syntax `` %`<constant_name>-<prefix> ``
+```yaml
+  constants:
+    diagnostic: 'Bundle.entry.resource.ofType(DiagnosticReport)[%orderIndex]'
+    rsext: '"https://reportstream.cdc.gov/fhir/StructureDefinition/"'
+  resource: '%resource.extension(%`rsext-parent-observation-value-descriptor`)'
+```
+
+* **For HL7 location spec**, all constants are treated as string replacements. The special syntax of
+`%{<constant_name>}` is used 
+```yaml
+  constants:
+     fieldPath: '/PATIENT_RESULT/ORDER_OBSERVATION(%{orderIndex})/OBR'
+  hl7Spec: [ '%{fieldPath}(%{entityIdIndex})-3' ]
+```
+
+# Extending Schemas
+Top level schemas can extend and existing top level schema in order to reduce duplicate elements and facilitate
+customization for different purposes. You can chain as many top level schemas as you wish. Note that reusing an element
+name will result in overwriting the existing element. 
```

Comment:
This is kind of where the confusion might set in, but I think it's helpful to elaborate on what elements can be overwritten.
```suggestion
name will result in overwriting the existing element, whether the existing element is at the top level of the extended schema or one of its child schemas. 
```